src/main/java/com/microsoft/sqlserver/jdbc/SQLServerColumnEncryptionAzureKeyVaultProvider.java

```diff
@@ -51,29 +55,35 @@
 
     private final static java.util.logging.Logger akvLogger = java.util.logging.Logger
             .getLogger("com.microsoft.sqlserver.jdbc.SQLServerColumnEncryptionAzureKeyVaultProvider");
+    private static final int KEY_NAME_INDEX = 4;
+    private static final int KEY_URL_SPLIT_LENGTH_WITH_VERSION = 6;
+    private static final String KEY_URL_DELIMITER = "/";
+    private HttpPipeline keyVaultPipeline;
+    private KeyVaultTokenCredential keyVaultTokenCredential;
+
     /**
      * Column Encryption Key Store Provider string
      */
     String name = "AZURE_KEY_VAULT";
 
-    private final String baseUrl = "https://{vaultBaseUrl}";
-
     private static final String MSSQL_JDBC_PROPERTIES = "mssql-jdbc.properties";
     private static final String AKV_TRUSTED_ENDPOINTS_KEYWORD = "AKVTrustedEndpoints";
+    private static final String RSA_ENCRYPTION_ALGORITHM_WITH_OAEP_FOR_AKV = "RSA-OAEP";
+
     private static final List<String> akvTrustedEndpoints;
-    static {
-        akvTrustedEndpoints = getTrustedEndpoints();
-    }
-    private final String rsaEncryptionAlgorithmWithOAEPForAKV = "RSA-OAEP";
 
     /**
      * Algorithm version
      */
     private final byte[] firstVersion = new byte[] {0x01};
 
-    private KeyVaultClient keyVaultClient;
+    private Map<String, KeyClient> cachedKeyClients = new ConcurrentHashMap<>();
+    private Map<String, CryptographyClient> cachedCryptographyClients = new ConcurrentHashMap<>();
+    private TokenCredential credential;
 
-    private KeyVaultCredential credentials;
+    static {
+        akvTrustedEndpoints = getTrustedEndpoints();
+    }
```

Comment:
Can just declare

`private static final List<String> akvTrustedEndpoints = getTrustedEndpoints();` since `getTrustedEndpoints` is static.